### PR TITLE
Ensure configuration examples are illustrative but still secure

### PR DESCRIPTION
**Pull request by JonZeolla**

### CODE REVIEW

**Review by chalin — COMMENTED** [review]
Can one of the @open-telemetry/collector-approvers comment on this?

File: content/en/docs/collector/configuration.md
Comment (Aneurysm9) [inline content/en/docs/collector/configuration.md:306]: Isn't `false` the default value?  If we're going to set this it should probably be removed instead.  That said, does this configuration actually work without the user taking steps to ensure a valid TLS cert on the other end?  Are those steps documented here?

Comment (JonZeolla) [reply content/en/docs/collector/configuration.md:306]: I was just keeping it to illustrate that it's an option that can be configured, while avoiding an insecure example

Comment (jpkrohling) [reply content/en/docs/collector/configuration.md:306]: The example as it is will fail, as it's not specifying which CA to use when doing validation of the server's certificate.

Comment (chalin) [reply content/en/docs/collector/configuration.md:306]: So is the vote to close this PR @jpkrohling @Aneurysm9? Or request further changes?

Comment (Aneurysm9) [reply content/en/docs/collector/configuration.md:306]: We should do one of those two things.  I'm not sure that I care overmuch which it is.

Comment (JonZeolla) [reply content/en/docs/collector/configuration.md:306]: I'm happy to add to the docs and specify an example CA bundle

Comment (JonZeolla) [reply content/en/docs/collector/configuration.md:306]: It looks like specifying a CA bundle is optional, but a cert and key is required for mTLS according to [this](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/jaegerexporter#getting-started) and [this](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md#tls--mtls-configuration).

Comment (jpkrohling) [reply content/en/docs/collector/configuration.md:306]: Specifying a CA bundle is optional, as the target server might have a cert signed by a known root CA. This is not the case for self-signed certs. I know this would make the guide longer, but I do think we should instruct people how to generate the certs for dev/testing purposes (hinting that they might want to find an appropriate way to generate/provision the production certs), and use those certs in the examples.

Otherwise, people will just copy/paste the insecure config into their production's config.

Comment (jpkrohling) [reply content/en/docs/collector/configuration.md:306]: If you need, use the steps I outlined in this blog post to generate the certs: https://medium.com/opentelemetry/securing-your-opentelemetry-collector-1a4f9fa5bd6f

**Comment by JonZeolla** [comment]
@austinlparker yes I will, hopefully this weekend or early next week

File: content/en/docs/collector/configuration.md
Comment (JonZeolla) [reply content/en/docs/collector/configuration.md:310]: @jpkrohling what do you think now?  I should have aligned it to your blog post, with some minor tweaks.

File: content/en/docs/collector/configuration.md
Comment (jpkrohling) [inline content/en/docs/collector/configuration.md:554]: How about something like this?

```suggestion
For a production setup, we strongly recommend using TLS certificates, either for secure communication or mTLS for mutual authentication.
```

File: content/en/docs/collector/configuration.md
Comment (jpkrohling) [inline content/en/docs/collector/configuration.md:555]:
```suggestion
See the below steps to generate self-signed certificates used in this example. You might want to use your current cert provisioning procedures to procure a certificate for production usage.
```

**Review by jpkrohling — COMMENTED** [review]
I have a small suggestion for the cert provisioning part, but LGTM.

File: content/en/docs/collector/configuration.md
Comment (chalin) [reply content/en/docs/collector/configuration.md:310]: I'd vote like @Aneurysm9 and remove `insecure: false` if it is the default.

File: content/en/docs/collector/configuration.md
Comment (chalin) [reply content/en/docs/collector/configuration.md:554]: I like your suggested change @jpkrohling.

File: content/en/docs/collector/configuration.md
Comment (chalin) [reply content/en/docs/collector/configuration.md:555]: I like your suggested change @jpkrohling.

**Comment by JonZeolla** [comment]
@jpkrohling I agree with your suggestions and merged them

**Comment by chalin** [comment]
@JonZeolla: can you remove the two occurrences of `insecure: false` -- https://github.com/open-telemetry/opentelemetry.io/pull/1162#discussion_r828524264.

**Comment by JonZeolla** [comment]
Sure

**Comment by austinlparker** [comment]
@JonZeolla is this ready to go?

**Comment by JonZeolla** [comment]
@austinlparker yup should be good!  Just pending a final approval.

I did look for the automerge option once I was comfortable but it doesn't seem to be enabled

**Review by jpkrohling — APPROVED** [review]

**Review by chalin — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
